sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSortExec.scala

```diff
@@ -418,7 +417,7 @@ case class GpuOutOfCoreSortIterator(
         // First we want figure out what is fully sorted from what is not
         val sortSplitOffset = if (pending.isEmpty) {
```

Comment:
do we need `sortSplitOffset`? Seems like instead it cold be replaced with `pending.isEmpty` down in the next if statement, and could remove this if statement.

Reply:
That would not cover the case where pending is not empty, but the next batch in pending does not overlap with what we just sorted.

Reply:
Got it, I missed the use right below of `sortSplitOffset`.